Objects/codeobject.c

```diff
@@ -94,6 +94,21 @@ intern_string_constants(PyObject *tuple)
     return modified;
 }
 
+static int
+code_check_buffer(PyObject *code) {
+    Py_buffer codebuffer = {};
+    if (!PyBytes_Check(code)) {
+        if (PyObject_GetBuffer(code, &codebuffer, PyBUF_SIMPLE))
+            return 0;
+
+        int contiguous = PyBuffer_IsContiguous(&codebuffer, 'C');
```

Comment:
This doesn't seem right. 
The buffer should be read-only and one dimensional, not just contiguous.

Reply:
Well tighten this up to:

codebuffer.readonly &&
                 codebuffer.ndim == 1 &&
                 codebuffer.strides == NULL;

I think that should satisfy the readonly check because per the documentation it needs to be consistent for all consumers: https://docs.python.org/3/c-api/buffer.html#c.PyBUF_WRITABLE

Reply:
For a well-behaved exporter, `PyBUF_SIMPLE` implies `C`, `ndim==1`, `B` and `readonly`.

I know the code base has these (redundant) checks in other places, but at some point we may want to use `assert()`.  I'm not suggesting to eliminate the extra safety check here and now, just clarifying.